torchvision/transforms/functional_tensor.py

```diff
@@ -915,7 +916,16 @@ def perspective(
     )
 
     ow, oh = img.shape[-1], img.shape[-2]
-    grid = _perspective_grid(perspective_coeffs, ow=ow, oh=oh, device=img.device)
+    dtype = _get_compatible_dtype(img)
+    grid = _perspective_grid(perspective_coeffs, ow=ow, oh=oh, dtype=dtype, device=img.device)
     mode = _interpolation_modes[interpolation]
 
     return _apply_grid_transform(img, grid, mode)
+
+
+def _get_compatible_dtype(img):
+    dtype = img.dtype if torch.is_floating_point(img) else torch.float32
+    if img.device.type == "cpu" and dtype == torch.float16:
+        # Avoid not implemented for 'Half' errors from "baddbmm__mkl" and "grid_sampler_2d_cpu_kernel_impl"
+        dtype = torch.float32
```

Comment:
In general, we used not to implement the float16 variants for CPU because they are just too slow. In order to be consistent with how PyTorch does things, let's just drop this case here, and adapt the tests to not test the `float16` codepath for CPU.

Reply:
Sounds good, let me remove those tests and drop this part of the code